torch/_dynamo/variables/tensor.py

```diff
@@ -644,6 +645,57 @@ def redistribute_fn_with_prim_types(x):
                 ),
                 **options,
             )
+        elif name == "register_hook":
+            # see [On tensor.register_hook]
+            assert len(args) == 1
+            fn_var = args[0]
+
+            if isinstance(fn_var, variables.NestedUserFunctionVariable):
+                # NestedUserFunctionVariable don't carry their fn, but reconstruction builds it
+                # This should not be onerous to support when needed.
+                unimplemented("NYI - lambda variables as hooks")
+            elif isinstance(fn_var.fn, functools.partial):
```

Comment:
This seems sus, don't you need to test what kind of VariableTracker you have before you try to access fields on it

Reply:
We can assume it one of the func ones, but yea, lets pop an assert in here